crates/compiler/builtins/bitcode/src/list.zig

```diff
@@ -159,13 +177,14 @@ pub const RocList = extern struct {
         element_width: usize,
     ) RocList {
         if (self.bytes) |source_ptr| {
-            if (self.isUnique()) {
-                if (self.capacity >= new_length) {
-                    return RocList{ .bytes = self.bytes, .length = new_length, .capacity = self.capacity };
+            if (self.isUnique() and !self.isSeamlessSlice()) {
+                const capacity = self.getCapacity();
```

Comment:
A quick optimization check idea: since we know at this point that it's not a seamless slice, is there any performance advantage in practice to using `self.capacity` directly here? (`self.getCapacity()` should ideally optimize to that in this case, but maybe it doesn't!)

Reply:
will do.